### PR TITLE
add custom fs loading

### LINKED ISSUE #56: Multiple static file packages
I'd like to produce two (or more) different statik fs's in different import paths. Due to relying on `init()` and no public API from the package, it's impossible to use both filesystems at the same time; concurrent access is also impossible as `fs.Register` is package-scoped.

I'd propose modifying the API:

`fs.New` would take `zipDatas ...string` and fallback on whatever was put in `fs.Register` (backwards compatible),
The generated filesystem package could add `Data() string` for this purpose in addition to keeping init() behaviour.

### PR DESCRIPTION
**Pull request by titpetric**
Closes #56 ; it only takes the first parameter for the FS data, but it would be trivial to support them all, I'm just not sure if it would be expected that they would be loaded one over the other.

### CODE REVIEW

**Comment by pavolloffay** [comment]
@rakyll hi, is this going to be merged?

File: fs/fs.go [hunk 1]
```diff
@@ -50,7 +50,10 @@ func Register(data string) {
 
 // New creates a new file system with the registered zip contents data.
 // It unzips all files and stores them in an in-memory map.
-func New() (http.FileSystem, error) {
+func New(zipDatas ...string) (http.FileSystem, error) {
```
Comment (rakyll) [inline fs/fs.go]: Wouldn't this be better with?

func New(zipdata []byte)

or

fun New(zipdata io.Reader)?

File: statik.go [hunk 2]
```diff
@@ -199,11 +199,14 @@ import (
 	"github.com/rakyll/statik/fs"
 )
 
-func init() {
-	data := "`, tags, comment, namePackage)
+func Data() string {
```
Comment (rakyll) [inline statik.go]: No need need to export Data.

File: fs/fs.go (hunk 1 above)
Comment (titpetric) [reply fs/fs.go]: Internally the codegen already stored the data in a string variable, didn't attempt to change. If you prefer another API here, let me know which. I prefer an io.Reader in this situation.

Comment (titpetric) [reply fs/fs.go]: (also, the variadic `...` is there to not break existing packages)

File: statik.go (hunk 2 above)
Comment (titpetric) [reply statik.go]: To create a new filesystem with fs.New from multiple generated imports inside the same project, the data needs to be accessible somehow. The use case is 1 filesystem for db migrations, and 1 filesystem for public_html static assets. Individual filesystems are created with `statikFS, err := fs.New(mysql.Data())` and another with `assets.Data()`...

File: fs/fs.go (hunk 1 above)
Comment (rakyll) [reply fs/fs.go]: Any signature change is a breaking change. For example:

```
var fn func() = statik.New
```

will break.

I'd add something like NewFromReader instead.

**Comment by pavolloffay** [comment]
@rakyll @titpetric could this be merged? We would like to start using this. Or we will have migrate to a different library.

**Comment by titpetric** [comment]
@pavolloffay please check goware/statik for a possible implementation that may suit you.

**Pull request closed without merging**